R/generatePartialDependence.R

```diff
@@ -246,378 +172,158 @@ generatePartialDependenceData = function(obj, input, features,
       target = td$class.levels
     else
       target = td$positive
-  }  else
+  } else
     target = "Risk"
 
-  args = list(obj = obj, data = data, fun = fun, td = td, individual = individual,
-    bounds = bounds, ...)
-
-  if (length(features) > 1L & !interaction) {
-    out = lapply(features, function(x) {
-      args$features = x
-      if (derivative) {
-        args$bounds = NULL
-        out = parallelMap(doPartialDerivativeIteration, x = rng[[x]], more.args = args)
-        rng = as.data.frame(rng[[x]])
-        colnames(rng) = x
-        centerpred = NULL
+  if (!derivative) {
+    args = list(model = obj, data = data, uniform = uniform, aggregate.fun = fun,
+      predict.fun = getPrediction, n = n, ...)
+    out = parallelMap(mmpf::marginalPrediction,
+      vars = if (interaction) list(features) else as.list(features), more.args = args)
+    if (length(target) == 1L) {
+      out = lapply(out, function(x) {
+        feature = features[features %in% names(x)]
+          names(x) = stri_replace_all(names(x), target, regex = "^preds")
+          x = data.table(x)
+          if (individual) {
+            x = melt(x, id.vars = feature, variable.name = "n", value.name = target)
+            x[, n := stri_replace(n, "", regex = target)]
+            setnames(x, c(feature, if (individual) "n" else "Function", target), names(x))
+          } else {
+            x
+          }
+        })
+    }
+  } else {
+    points = sapply(features, function(x) uniformGrid(data[[x]], n[1]), simplify = FALSE)
```

Comment:
We try to not use `sapply` -- could this be `lapply` instead?

Reply:
the reason i am using it is so that i get a named list. is there an alternative way to do this that isn't terribly verbose?

Reply:
There's a few options here: https://stackoverflow.com/questions/9469504/access-and-preserve-list-names-in-lapply-function